pkg/v1/tkg/tkgpackageclient/package_update.go

```diff
@@ -55,31 +59,25 @@ func (p *pkgClient) UpdatePackage(o *tkgpackagedatamodel.PackageOptions, progres
 	}
 
 	pkgInstallToUpdate := pkgInstall.DeepCopy()
-	if o.Version != pkgInstallToUpdate.Status.Version {
-		if pkgInstallToUpdate.Spec.PackageRef == nil || pkgInstallToUpdate.Spec.PackageRef.VersionSelection == nil {
-			err = errors.New(fmt.Sprintf("failed to update package '%s'", o.PkgInstallName))
-			return
-		}
-		progress.ProgressMsg <- fmt.Sprintf("Getting package metadata for '%s'", pkgInstallToUpdate.Spec.PackageRef.RefName)
-		o.PackageName = pkgInstallToUpdate.Spec.PackageRef.RefName
-		if _, _, err = p.GetPackage(o); err != nil {
-			return
-		}
-		pkgInstallToUpdate.Spec.PackageRef.VersionSelection.Constraints = o.Version
+
+	if pkgInstallToUpdate.Spec.PackageRef == nil || pkgInstallToUpdate.Spec.PackageRef.VersionSelection == nil {
+		err = errors.New(fmt.Sprintf("failed to update package '%s' as no existing version was found in the package install", o.PkgInstallName))
+		return
 	}
 
-	if o.ValuesFile != "" {
-		if secretCreated, err = p.updateValuesFile(o, pkgInstallToUpdate, progress.ProgressMsg); err != nil {
-			return
-		}
+	o.PackageName = pkgInstallToUpdate.Spec.PackageRef.RefName
+	if o.Version != "" {
+		pkgInstallToUpdate.Spec.PackageRef.VersionSelection.Constraints = o.Version
+	} else {
+		o.Version = pkgInstallToUpdate.Spec.PackageRef.VersionSelection.Constraints
```

Comment:
This line can be confusing. Probably add a comment to say pass the version back and will be used to validate if package exists in line 75.

Reply:
Thanks Lucheng for this review, reminded me to get rid of Get Package() call